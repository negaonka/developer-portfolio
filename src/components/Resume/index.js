import React from 'react'
import resume from '../../assets/data/ResumeNehaGaonkar.pdf';

function Resume(){
    const frontEndList = [{name: 'JavaScript'}, {name: 'JQuery'}, {name: 'Ajax'}, {name: 'Angular'}, {name: 'HTML'}, {name:'CSS'}, {name: 'React'}];
    const backEndList = [{name: 'Java'}, {name: 'Node'}, {name: 'Hibernate'}, {name: 'Web-services'},{name: 'Oracle'}, {name: 'MY-SQL'}];
    return(
        <>
        <section>
        <h1 data-testid="h1tag">Resume</h1>
        <div>
        <h5>Download my<a style={{color:"#86868b"}} href={resume} download="ResumeNehaGaonkar.pdf"><h5>resume</h5></a></h5>
        </div>
        </section>
        <section>
            <h2>UI Technologies</h2>
            <ul>
            {frontEndList.map((technology) => (
            <li style={{listStyle: "circle"}} key={technology.name}>{technology.name}</li> 
            ))}
            </ul>
            <h2>BackEnd Technologies</h2>
            <ul>
            {backEndList.map((technology) => (
            <li style={{listStyle: "circle"}} key={technology.name}>{technology.name}</li> 
            ))}
            </ul>
        </section>
        </>
    )
}

export default Resume