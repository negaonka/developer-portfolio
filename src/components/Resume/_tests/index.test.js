import React from 'react';
import { render, cleanup } from '@testing-library/react';
import '@testing-library/jest-dom/extend-expect';
import Resume from '..';

afterEach(cleanup);  //to clean all the memory data

describe('Project component', () => {
    // baseline test
    it('renders', () => {
        render(<Resume/>);
      });
   
    // snapshot test
    it('matches snapshot', () => {
        const { asFragment } = render(<Resume/>);
        // assert value comparison
        expect(asFragment()).toMatchSnapshot();
      });

    })

     //text visibility
     describe('texts are visible', () => {
        it('inserts text', () => {
          // Arrange
          const { getByTestId } = render(<Resume/>);
          // Assert
          expect(getByTestId('h1tag')).toHaveTextContent('Resume');

        });
      })