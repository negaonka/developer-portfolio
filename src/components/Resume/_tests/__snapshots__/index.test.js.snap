// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Project component matches snapshot 1`] = `
<DocumentFragment>
  <section>
    <h1
      data-testid="h1tag"
    >
      Resume
    </h1>
    <div>
      <h5>
        Download my
        <a
          download="ResumeNehaGaonkar.pdf"
          href="ResumeNehaGaonkar.pdf"
          style="color: rgb(134, 134, 139);"
        >
          <h5>
            resume
          </h5>
        </a>
      </h5>
    </div>
  </section>
  <section>
    <h2>
      UI Technologies
    </h2>
    <ul>
      <li
        style="list-style: circle;"
      >
        JavaScript
      </li>
      <li
        style="list-style: circle;"
      >
        JQuery
      </li>
      <li
        style="list-style: circle;"
      >
        Ajax
      </li>
      <li
        style="list-style: circle;"
      >
        Angular
      </li>
      <li
        style="list-style: circle;"
      >
        HTML
      </li>
      <li
        style="list-style: circle;"
      >
        CSS
      </li>
      <li
        style="list-style: circle;"
      >
        React
      </li>
    </ul>
    <h2>
      BackEnd Technologies
    </h2>
    <ul>
      <li
        style="list-style: circle;"
      >
        Java
      </li>
      <li
        style="list-style: circle;"
      >
        Node
      </li>
      <li
        style="list-style: circle;"
      >
        Hibernate
      </li>
      <li
        style="list-style: circle;"
      >
        Web-services
      </li>
      <li
        style="list-style: circle;"
      >
        Oracle
      </li>
      <li
        style="list-style: circle;"
      >
        MY-SQL
      </li>
    </ul>
  </section>
</DocumentFragment>
`;
